### DISCUSSION TITLE
How to run selective requests within a collection.

### DISCUSSION THREAD

**Discussion by rudrajyotib** [usebruno/bruno#2460] [Q&A]
I have a collection with some requests in sequence which is used for running a full set of tests. But, there are times, when I need to run only a part of them. Is there a way to run selective requests within a collection or a folder in it?

**Comment by csibug** [comment]
https://docs.usebruno.com/scripting/javascript-reference#request-order

**Comment by rudrajyotib** [comment]
@csibug I want to do it from UI.

**Reply by rreyn-bruno** [reply]
Hi @rudrajyotib - I realize it's been quite a while, but we released this capability some time ago. Have you had a chance to give it a try and see if it suits your use case? 

<img width="1051" height="846" alt="image" src="https://github.com/user-attachments/assets/bc0155f4-1e1b-4d4a-8a18-28727e94be02" />

**Comment by jocelynduc** [comment]
Hi all,

I think I have the same need, if I correctly understood @rudrajyotib.

Running a **selection** of requests from a folder with the "Run" feature
How can I achieve that with Bruno UI? Due to the number of requests and use cases I have, tags are not an option for me

<img width="868" height="818" alt="image" src="https://github.com/user-attachments/assets/4a8c0572-a32b-49ec-8644-cf6fe38e12e2" />

**Reply by rreyn-bruno** [reply]
Hi @jocelynduc - Bruno includes a similar functionality. In v3.3.0, we've updated the UI to make this more clear. Could you take a look at see if this resolves your issue? 

<img width="1051" height="846" alt="image" src="https://github.com/user-attachments/assets/1b84cfce-5b87-4143-b225-6d3614e2f0e4" />

**Reply by jocelynduc** [reply]
Hi @rreyn-bruno,
Thanks for the tips, I tried with 3.3.0. Yes, I can now achieve a "selective" run on a collection, but not at folder level. I think this feature, which I need, is not available yet.

Right click on a collection, Run => I can select the request with checkboxes
Right click on a folder, Run => there is no list, no checkbox ;)

**Reply by timolex** [reply]
Dear all, please check out my pull request https://github.com/usebruno/bruno/pull/8346 implementing selective Folder Runs. Cheers from Winterthur, Switzerland 😎
